testing/architecture-test/src/test/java/org/gradle/architecture/test/ProviderMigrationArchitectureTest.java

```diff
@@ -206,29 +210,29 @@ public boolean test(JavaClass input) {
         .andShould().notBeAnnotatedWith(NotToBeReplacedByLazyProperty.class));
 
     private static HaveLazyReturnType haveProviderReturnType() {
-        return new HaveLazyReturnType(Property.class, Provider.class);
+        return new HaveLazyReturnType(Arrays.asList(Property.class, MapProperty.class, ListProperty.class, SetProperty.class), Collections.singletonList(Provider.class));
```

Comment:
This is planned to the master branch directly, right?

Reply:
I can extract it into a separate PR, no problem

Reply:
@asodja PTAL 
https://github.com/gradle/gradle/pull/29805